compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -387,6 +387,18 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
             ty.needs_drop(self.fcx.tcx, self.fcx.param_env)
         };
 
+        let find_parent_expr = |mut hir_id| {
+            let hir = self.fcx.tcx.hir();
+            hir_id = hir.find_parent_node(hir_id)?;
+            loop {
+                if let hir::Node::Expr(_) = self.fcx.tcx.hir().find(hir_id)? {
```

Comment:
```suggestion
                if let hir::Node::Expr(_) = hir.find(hir_id)? {
```

Reply:
Ah, I meant to do that when I was writing this and forgot! At any rate, I completely rewrote this code using `parent_iter`.